test/e2e/app-dir/action-in-pages-router/action-in-pages-router.test.ts

```diff
@@ -0,0 +1,37 @@
+import { nextTestSetup } from 'e2e-utils'
+import { assertNoRedbox, retry } from 'next-test-utils'
+
+describe('app-dir - action-in-pages-router', () => {
+  const { next, isNextStart } = nextTestSetup({
+    files: __dirname,
+  })
+
+  it('should not error on fake server action in pages router', async () => {
+    const browser = await next.browser('/foo')
+    const button = await browser.elementByCss('button')
+    await button.click()
+
+    await retry(async () => {
+      const browserLogText = (await browser.log())
+        .map((item) => item.message)
+        .join('')
+      // This is a fake server action, a simple function so it can still work
+      expect(browserLogText).toContain('action:foo')
+      await assertNoRedbox(browser)
+    })
+  })
+
+  if (isNextStart && !process.env.TURBOPACK) {
```

Comment:
we should clarify why we're disabling the test for turbopack here or add it to the existing manifest